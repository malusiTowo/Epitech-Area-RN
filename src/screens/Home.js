import React, { useRef, useState, useContext, useEffect } from 'react'
import { View, ScrollView, StyleSheet, TextInput, StatusBar, Image, RefreshControl, Dimensions } from 'react-native'
import { Button, Text } from 'native-base'
import { Entypo } from '@expo/vector-icons'
import RBSheet from "react-native-raw-bottom-sheet";
import { observer } from 'mobx-react';
const { width } = Dimensions.get('screen');

import defaultServices from '../config/defaultServices';
import ThinWidget from '../components/widgets/ThinWidget';
import SuggestionWidget from '../components/widgets/SuggestionWidget';
import { StoreContext } from '../store/Context';
import { getUserServices } from '../api/Services';
import { getDeviceId } from '../api/Auth'

const Home = (props) => {
  const [defaultWidgets, setDefaultWidgets] = useState(defaultServices);
  const [refreshing, setRefreshing] = useState(false);
  const [services, setServices] = useState([]);
  const { name, url } = useContext(StoreContext);

  const handleGetUserServices = async () => {
    try {
      const services = await getUserServices(url);
      setServices(services);
    } catch (err) {
      console.log("err", err);
    }
  }

  const handleSendDeviceId = async () => {
    try {
      await getDeviceId(url);
    } catch (err) {
      console.log(err);
    }
  }

  useEffect(() => {
    handleSendDeviceId();
    handleGetUserServices();
  }, []);

  const onRefresh = React.useCallback(() => {
    setRefreshing(true);
    handleGetUserServices()
    setRefreshing(false);
  }, [refreshing]);
  let RbSheet = useRef(null);


  const closeRbSheet = () => RbSheet.close();

  const goToCreateWidget = async () => {
    closeRbSheet();
    setTimeout(() => {
      props.navigation.navigate('CreateWidget')
    }, 500);
  }

  const searchWidget = txt => {
    const input = txt.trim().toLowerCase();
    if (input.length === 0) return setDefaultWidgets(defaultServices);

    const widgets = defaultServices.filter(({ title }) => title.toLowerCase().search(input) > -1);
    setDefaultWidgets(widgets);
  }

  return (
    <View style={styles.container}>
      <StatusBar backgroundColor="white" barStyle="dark-content" />

      <View style={{
        flexDirection: 'row',
        justifyContent: 'space-between',
        height: 130,
        alignItems: 'center',
        width: '100%',
        marginTop: 10
      }} >
        <Image style={{ width: 140, height: 70, marginLeft: 20 }} source={require('../assets/ifttt_logo.png')} />
        <Text style={{ fontSize: 20, fontWeight: '500', marginRight: 20 }}
          onPress={() => props.navigation.navigate('Profile')} >
          {name ? name : 'Settings'}
        </Text>
      </View>

      <ScrollView
        horizontal={false}
        style={styles.list}
        refreshControl={<RefreshControl refreshing={refreshing} onRefresh={onRefresh} />}
      >
        {
          services.map((service, idx) => (
            <View key={idx} style={styles.listItem}>
              <ThinWidget
                icon={service.icon}
                text={service.text || service.title}
                backgroundColor={service.backgroundColor || service.color || 'red'}
                navigation={props.navigation}
                serviceId={service.id}
              />
            </View>
          ))
        }

      </ScrollView>

      <View style={styles.btnWrapper}>
        <Button onPress={() => RbSheet.open()} style={styles.btn}>
          <Text style={styles.btnText}>Get more</Text>
        </Button>
      </View>

      <RBSheet
        ref={ref => RbSheet = ref}
        height={600}
        animationType="fade"
        closeOnDragDown
        customStyles={{
          container: styles.rbSheetContainer,
          wrapper: styles.rbSheetWrapper,
          draggableIcon: styles.rbSheetDragIcon
        }}
      >
        <ScrollView>
          <View style={styles.inputWrapper}>
            <TextInput
              placeholder="Search"
              style={styles.input}
              placeholderTextColor="#ddd"
              onChangeText={searchWidget}
            />
          </View>
          <View style={styles.diyWrapper}>
            <Text style={styles.diyText}>Make your own Applets from scratch</Text>
            <Button onPress={goToCreateWidget} style={styles.diyBtn}>
              <Entypo name="plus" size={25} color="white" />
            </Button>
          </View>

          {
            defaultWidgets.map(({ title, iconOne, iconTwo, color, data }, idx) => {
              return (
                <View style={styles.rbListItem} key={idx}>
                  <SuggestionWidget
                    closeRbSheet={closeRbSheet}
                    navigation={props.navigation}
                    text={title}
                    iconOne={iconOne}
                    iconTwo={iconTwo}
                    backgroundColor={color}
                    data={data}
                  />
                </View>
              )
            })
          }

        </ScrollView>
      </RBSheet>
    </View>
  )
}

Home.navigationOptions = ({ navigation }) => ({
  headerShown: false,
});


const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    flex: 1,
    backgroundColor: '#fff'
  },
  list: {
    flex: 1,
  },
  listItem: {
    marginVertical: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.4,
    shadowRadius: 2,
    elevation: 1,
  },
  btnWrapper: {
    backgroundColor: 'black',
    height: 120,
    bottom: 0,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center'
  },
  btn: {
    backgroundColor: 'grey',
    borderRadius: 25,
    marginTop: 20,
    justifyContent: 'center'
  },
  btnText: {
    color: 'white',
    fontSize: 20,
    fontWeight: 'bold',
    width: 150,
    textAlign: 'center'
  },
  // behind of modal
  rbSheetWrapper: {
  },
  // background of modal
  rbSheetContainer: {
    // paddingTop: 20
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10
  },
  rbSheetDragIcon: {
    backgroundColor: 'black'
  },
  rbListItem: {
    marginVertical: 10,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.4,
    shadowRadius: 2,
    elevation: 1,
  },
  diyWrapper: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-around',
    width: width - 20,
  },

  diyText: {
    fontSize: 13,
    fontWeight: '700',
    color: 'black',
    padding: 20,
  },
  diyBtn: {
    backgroundColor: 'black',
    borderRadius: 20,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    width: 90,
    height: 35,
    alignSelf: 'center'
  },
  inputWrapper: {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  input: {
    borderRadius: 10,
    borderColor: '#ccc',
    borderWidth: 3,
    paddingLeft: 5,
    width: '90%',
    marginTop: 25,
    height: 60,
    fontSize: 18,
    paddingLeft: 10,
    color: 'black',
    fontWeight: 'bold'
  },
})

export default observer(Home);
