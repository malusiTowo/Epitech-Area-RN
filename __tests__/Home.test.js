import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../src/screens/Home';

it('Home renders correctly', () => {
  const home = renderer.create(<Home />).toJSON();
  expect(home).toMatchSnapshot();
});
