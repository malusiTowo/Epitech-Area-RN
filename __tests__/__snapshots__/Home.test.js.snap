// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#007aff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 5,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      uppercase={false}
    >
      Go to profile
    </Text>
  </View>
  <RCTScrollView
    horizontal={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "elevation": 1,
            "marginVertical": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 2,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 7,
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 80,
                  "width": "90%",
                },
                Object {
                  "backgroundColor": "orange",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "marginRight": 20,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Date & Time
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "elevation": 1,
            "marginVertical": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 2,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 7,
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 80,
                  "width": "90%",
                },
                Object {
                  "backgroundColor": "tomato",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "marginRight": 20,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              IOS Photos
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "elevation": 1,
            "marginVertical": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 2,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 7,
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 80,
                  "width": "90%",
                },
                Object {
                  "backgroundColor": "black",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "marginRight": 20,
                }
              }
            >
              <Text />
            </View>
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Alexa wake-up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "display": "flex",
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "grey",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 25,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "width": 120,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
      >
        Get more
      </Text>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#00000077",
              "flex": 1,
            },
            Object {},
          ],
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "opacity": 1,
          }
        }
      />
      <View
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "height": 0,
            "overflow": "hidden",
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#ccc",
                  "borderRadius": 5,
                  "height": 5,
                  "margin": 10,
                  "width": 35,
                },
                Object {
                  "backgroundColor": "black",
                },
              ]
            }
          />
        </View>
        <RCTScrollView>
          <View>
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                onChangeText={[Function]}
                placeholder="Search"
                placeholderTextColor="#ddd"
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderColor": "#ccc",
                    "borderRadius": 10,
                    "borderWidth": 3,
                    "color": "black",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "height": 60,
                    "marginTop": 25,
                    "paddingLeft": 10,
                    "width": "90%",
                  }
                }
                underlineColorAndroid="transparent"
              />
            </View>
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "black",
                      "fontSize": 13,
                      "fontWeight": "700",
                      "padding": 20,
                    },
                  ]
                }
                uppercase={false}
              >
                Make your own Applets from scratch
              </Text>
              <View
                accessible={true}
                focusable={false}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "black",
                    "borderBottomWidth": null,
                    "borderColor": "#007aff",
                    "borderLeftWidth": null,
                    "borderRadius": 20,
                    "borderRightWidth": null,
                    "borderTopWidth": null,
                    "borderWidth": undefined,
                    "elevation": 2,
                    "flexDirection": "column",
                    "height": 35,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": undefined,
                    "shadowOffset": undefined,
                    "shadowOpacity": undefined,
                    "shadowRadius": undefined,
                    "width": 90,
                  }
                }
              >
                <Text />
              </View>
            </View>
            <View
              style={
                Object {
                  "elevation": 1,
                  "marginVertical": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.4,
                  "shadowRadius": 2,
                }
              }
            >
              <View
                accessible={true}
                focusable={false}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "black",
                      "borderRadius": 7,
                      "height": 150,
                      "width": "95%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "margin": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "fontSize": 25,
                          "fontWeight": "bold",
                          "padding": 20,
                        }
                      }
                    >
                      Get the weather forecast every day at 7:00 AM
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "display": "flex",
                        "flexDirection": "row",
                        "justifyContent": "flex-end",
                        "marginBottom": 15,
                        "paddingBottom": 20,
                      }
                    }
                  >
                    <View>
                      <Text />
                    </View>
                    <View
                      style={
                        Object {
                          "marginHorizontal": 10,
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "elevation": 1,
                  "marginVertical": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.4,
                  "shadowRadius": 2,
                }
              }
            >
              <View
                accessible={true}
                focusable={false}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "black",
                      "borderRadius": 7,
                      "height": 150,
                      "width": "95%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "margin": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "fontSize": 25,
                          "fontWeight": "bold",
                          "padding": 20,
                        }
                      }
                    >
                      Get the weather forecast every day at 7:00 AM
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "display": "flex",
                        "flexDirection": "row",
                        "justifyContent": "flex-end",
                        "marginBottom": 15,
                        "paddingBottom": 20,
                      }
                    }
                  >
                    <View>
                      <Text />
                    </View>
                    <View
                      style={
                        Object {
                          "marginHorizontal": 10,
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "elevation": 1,
                  "marginVertical": 10,
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 0.4,
                  "shadowRadius": 2,
                }
              }
            >
              <View
                accessible={true}
                focusable={false}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "black",
                      "borderRadius": 7,
                      "height": 150,
                      "width": "95%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "margin": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "white",
                          "fontSize": 25,
                          "fontWeight": "bold",
                          "padding": 20,
                        }
                      }
                    >
                      Get the weather forecast every day at 7:00 AM
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "display": "flex",
                        "flexDirection": "row",
                        "justifyContent": "flex-end",
                        "marginBottom": 15,
                        "paddingBottom": 20,
                      }
                    }
                  >
                    <View>
                      <Text />
                    </View>
                    <View
                      style={
                        Object {
                          "marginHorizontal": 10,
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
